### PR TITLE
Dockerfile: fix envsubst dependency

### PR DESCRIPTION
**Pull request by fromanirh**
fix non-OCP builds by bundling a local and specialized version of envsubst, which can do partial variable substitution.

Signed-off-by: Francesco Romani <fromani@redhat.com>

### CODE REVIEW

**Comment by fromanirh** [comment]
/hold
actually bad idea, need more refinement

**Comment by fromanirh** [comment]
/hold cancel
I'm now happy enough with this PR

File: hack/run-test-serial-e2e.sh [hunk 1]
```diff
@@ -29,6 +27,67 @@ if [ -n "${E2E_SERIAL_SKIP}" ]; then
...
+		*)
+			echo "unrecognized option: $1"
+			echo "use the form --opt val and not the form --opt=val"
+			exit 1
+			;;
+	esac
+done
+
+# mandatory
+if [ -z "$REPORT_DIR" ]; then
```
Comment (shajmakh) [inline hack/run-test-serial-e2e.sh:81]: I am not familiar with shell lang, but I see all the vars are used within {}, do we need them to wrap REPORT_DIR?

**Comment by fromanirh** [comment]
infra issue (limits exceeded)

**Review by jlojosnegros — APPROVED** [review]
/lgtm

File: hack/envsubst.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/drone/envsubst"
```
Comment (cynepco3hahue) [inline hack/envsubst.go:9]: Why can not we just use shell `envsubst`?

Comment (fromanirh) [reply hack/envsubst.go:9]: It's an option. We will need this in the builder image (which is debian based though), and we need a specific behaviour (= replace only the env vars which have an actual value set) which is a bit harder to guarantee using shell tools.
I think all in all this approach has enough merit to be proposed, but I'm not pushing for it so if we all agree to use the shell variant (provided the above requirements are met) we can go this way.

**Pull request merged**